keyring/keyring.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018-2019 gopcua authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+// Package keyring implements a simple key-value store for certificates,
+// public and private RSA keys, and the certificate thumbprints.
+// It is meant to be an internal use only package to the gopcua library.
+package keyring
+
+import (
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/hex"
+	"errors"
+	"sync"
+)
+
+type privateKeys struct {
+	thumbprint []byte
+	cert       *x509.Certificate
+	key        *rsa.PrivateKey
+}
+
+var localKeys sync.Map
```

Comment:
I suggest to make this a `struct Keyring` and the function as methods. Global variables make testing harder.

Reply:
Yea, that bothered me from the start.  I'm debating if this is actually required at all.  I'll put this in the securitypolicy library and try to find a good spot for this struct.